src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlDependency.cs

```diff
@@ -597,11 +605,13 @@ private static void ObtainProcessDispatcher()
                             _processDispatcher = dependency.SingletonProcessDispatcher; // Set to static instance.
 
                             // Serialize and set in native.
-                            ObjRef objRef = GetObjRef(_processDispatcher);
-                            BinaryFormatter formatter = new BinaryFormatter();
-                            MemoryStream stream = new MemoryStream();
-                            GetSerializedObject(objRef, formatter, stream);
-                            SNINativeMethodWrapper.SetData(stream.GetBuffer()); // Native will be forced to synchronize and not overwrite.
+                            using (MemoryStream stream = new MemoryStream())
+                            {
+                                SqlClientObjRef objRef = new SqlClientObjRef(_processDispatcher);
+                                DataContractSerializer formatter = new DataContractSerializer(objRef.GetType());
```

Comment:
Let's rename `formatter` where it's replaced by a `DataContractSerializer`.

Reply:
usually formatter comes from IFormatter. I did not want to change that, but if you insist I can rename it to serializer.